cylc/sphinx_ext/cylc_lang/domains.py

```diff
@@ -328,14 +328,24 @@ def sanitise_signature(cls, sig):
         Examples:
             >>> CylcSettingDirective.sanitise_signature('a=b')
             ('a', 'b')
+            >>> CylcSettingDirective.sanitise_signature('share/cycle')
+            ('share/cycle', None)
 
         """
         value = None
+        is_slash_in_sig = False
+        # This is necessary to allow slash config entries, e.g. share/cycle
+        if "/" in sig:
+            is_slash_in_sig = True
+            sig = sig.replace('/', 'FORWARD_SLASH')
+
         if cls.NAME == 'setting':
             tokens = tokenise(sig)
             value = tokens['value']
             tokens['value'] = None
             sig = detokenise(tokens)
+            if is_slash_in_sig is True:
+                sig = sig.replace('FORWARD_SLASH', '/')
         return sig, value
```

Comment:
Non-settings will be left with `FORWARD_SLASH` in their sig?

I think this is roughly equivalent to `value.replace('/', 'FORWARD_SLASH')`.

Reply:
I've de-dented to catch the non-settings - sorry I didn't spot that!

Reply:
So now it's exactly equivalent to `value.replace('/', 'FORWARD_SLASH')`.

Ah, brain cogs turning, does the tokeniser produce traceback for `/` chars?

Reply:
Yes needs to have the slash removed before going into tokeniser - fails with ValueError namespace problems there. Which is why I had initially suggested to adapt tokeniser by adding a regex to have `/` on the conditions for accepted strings. 
So what this has to do is remove the `/` and replace with an accepted string (hence, the `FORWARD_SLASH`), then after it has been through it, it needs to then put the slash back so it can appear correctly in the docs. Feels ever so slightly inelegant, but avoids changing tokeniser, as you suggested to leave that bit of code alone.

Reply:
Ah yes, it's slowly coming back to me!

Reply:
Er, unfortunately this addition to the functional test flags an issue:

```
--- a/cylc/sphinx_ext/cylc_lang/__init__.py
+++ b/cylc/sphinx_ext/cylc_lang/__init__.py
@@ -103,6 +103,12 @@ Sphinx domain for ``cylc`` configurations.
 
               an iso8601 duration.
 
         .. cylc:setting:: a/b/c
 
            A setting with a ``/`` in the name.
 
+        See also :cylc:conf:`a/b/c`.
 
```

```
  File "/Users/oliver/cylc-sphinx-extensions/cylc/sphinx_ext/cylc_lang/domains.py", line 126, in tokenise
    raise ValueError(f'Not a valid namespace "{namespace_string}"')
ValueError: Not a valid namespace "a/b/c"
The full traceback has been saved in /var/folders/rg/691l5s513_97d5s7ktr1rtt00000gn/T/sphinx-err-o343iiwg.log, if you want to report the issue to the developers.
```

Don't need to touch the tokeniser, but I think we might need to add `/` as a valid character in `CYLC_WORD`.

Reply:
The good news is that `/` chars in URL fragments seem to be fine with Sphinx.